### ISSUE TITLE
Formalize namespace assignment

### ISSUE THREAD

**Issue by kpreid** [gnuradio/SigMF#20]
I was asked to review the SigMF specification by @bhilburn.

It currently says nothing whatsoever about how non-core namespace prefixes are managed. I suggest that it should do so.

----

I have a concrete proposal: reuse the XML Namespaces standard in its entirety, as if SigMF name-value pairs were XML attributes. (Yes, there's plenty that's awful in XML-land and should not be imitated, but this is not one of those things, _given the premise that we want namespaces at all._)

In brief, what that standard says is that a fully-qualified name is a pair of a _URI_ and a local name. The URI serves to identify the namespace; thus the question of "who gets to define this namespace" is delegated to the systems we already use for interpreting URIs. (This __does not__ mean that a SigMF application is ever required to attempt to download anything via that URI.)

In order to make it convenient to use, a URI is aliased to a local name using an extra attribute. If we copy XML exactly without even renaming anything, this might look like the following, in an example where we had a "description with HTML" namespace as I proposed in a later comment in #7:

```
{
    "global": {
        "xmlns:withhtml": "http://switchb.org/ns/sigmf-with-html"
        "description": "Wow!",
        "withhtml:description": "<a href='https://en.wikipedia.org/wiki/Wow!_signal'>Wow!</a>"
    }
}
```

(I am omitting, for now, the question of how the `core:` prefix works, and whether `xmlns=` is allowed, and showing only an extension with an explicit prefix.)

The `xmlns` attributes (er name-value pairs) can be on any parent node of the usage site. The prefix `withhtml:` here is an alias defined within the document; the extension does not get to say what prefix is used.

If it offends someone's aesthetic, we can use a different string instead of `xmlns` (with infinitesimal compatibility issues).

----

Reusing XML Namespaces would also allow SigMF documents to reuse assorted preexisting metadata schemas, where they fit well, rather than defining totally new vocabulary. I'm going to say "Semantic Web" and "RDF" here for people who know what I mean, and then note again that _reusing the vocabulary does not mean you have to buy into the whole thing_ and this is a nicely separable piece.

This also can be done without borrowing the exact syntax of XML Namespaces; all that is required is the notion that a fully-qualified name is/maps-to a pair of a URI and a local name.

**Comment by bhilburn** [comment]
Just to make sure I'm looking at the right thing, are you referring to this?
https://www.w3.org/TR/REC-xml-names/

**Comment by kpreid** [comment]
That looks right. I'm not sure whether anything has superseded it or not but it should be fine for discussion.

**Comment by bhilburn** [comment]
Closing this one out, as it will be implemented using #108 in v0.1.0 of the specification.
